static/js/narrow.js

```diff
@@ -125,40 +125,46 @@ export function set_narrow_title(title) {
 }
 
 function update_narrow_title(filter) {
-    const filter_title = filter.get_title();
-    const search_default = $t({defaultMessage: "Search results"});
-
-    if (filter_title !== undefined) {
-        if (filter.has_operator("stream")) {
-            if (!filter._sub) {
-                // The stream is not set because it does not currently
-                // exist (possibly due to a stream name change), or it
-                // is a private stream and the user is not subscribed.
-                set_narrow_title(filter_title);
-            } else if (filter.has_operator("topic")) {
-                const topic_name = filter.operands("topic")[0];
-                set_narrow_title("#" + filter_title + " > " + topic_name);
-            } else {
-                set_narrow_title("#" + filter_title);
-            }
-        } else if (filter.has_operator("pm-with")) {
-            const emails = filter.operands("pm-with")[0];
-            const user_ids = people.emails_strings_to_user_ids_string(emails);
-            if (user_ids !== undefined) {
-                const names = people.get_recipients(user_ids);
-                set_narrow_title(names);
-            } else {
-                if (emails.includes(",")) {
-                    set_narrow_title("Invalid users");
+    if (filter !== undefined) {
```

Comment:
It would likely be better to use the early return pattern rather than putting all the existing code of this function inside a big if/else block. Might be a sensible refactor for this whole function, to be honest.